content/articles/multi-label-classification-with-scikit-multilearn/index.md

```diff
@@ -0,0 +1,654 @@
+
+#### Label powerset
+
+It transforms the problem into a multi-class problem. Each multi-class classifier is then trained with unique label combinations found in the data.
+
+The goal label powerset is to find a combination of unique labels and assign them different values.
+
+Let's see an example.
+
+![Label powerset](/engineering-education/multi-label-classification-with-scikit-multilearn/label-powerset.png)
+
+In this example we have three unique label combination, it is then assigned values `1`, `2` and `3`. The multi-class classifier is then trained on all three unique label combinations.
+This technique tends to give higher accuracy.
+
+### Adapted algorithm
+
+This technique uses adaptive algorithms, which are used to perform multi-label classification rather than conducting problem transformation directly.
+In Scikit-multi learn, we have multi-label-k-nearest-neighbor (MLkNN), which is used to handle multi-label classification.
+
+### Ensemble methods
+
+This is a hybrid technique that combines the functionalities of the two mentioned techniques to produce better results. Ensemble methods tend to give a model with a higher accuracy score.
+
+For a detailed understanding about ensemble methods click [here](/engineering-education/boosting-algorithms-python/)
+
+> NOTE: In this tutorial, we will focus on the first technique, problem transformation, in building our model.
+
+Before we start building our model, let's see the dataset that we will be using.
+
+### Dataset
+
+As mentioned earlier, we will build a multi-label text classification, and the model will be able to classify given text into different categories. The categories in our dataset are `mysql`, `python`, and `php`, a text can belong to one or more of the given categories. If a text belongs to a certain category it is assigned `1`; if it does not, it is assigned `0` as shown in the image below.
+
+![Snip of the dataset used](/engineering-education/multi-label-classification-with-scikit-multilearn/dataset-snip.jpg)
+
+To download the CSV file for this dataset click [here](https://drive.google.com/file/d/1_scJyWek8DJUZ14_wZ8GXZslpwT8ubzt/view?usp=sharing). After downloading the dataset, name it as `dataset-tags.csv`.
+
+### Loading exploratory data analysis packages
+
+Exploratory data analysis(EDA) packages are used for data analysis and data manipulation. We shall use pandas to read our dataset and Numpy to perform mathematical computations.
+
+```python
+import pandas as pd
+import numpy as np
+```
+
+### Loading dataset
+
+We use pandas to load our dataset.
+
+```python
+df = pd.read_csv("dataset-tags.csv")
+```
+
+### Checking data structure
+
+We check the structure to be able to see the available columns in our dataset.
+
+```python
+df.head()
+```
+
+The output is as shown.
+
+![Dataset structure](/engineering-education/multi-label-classification-with-scikit-multilearn/dataset-structure.jpg)
+
```

Comment:
What do we decode from these results? Explain to the reader what these outputs mean